### PR TITLE
btcm: add order execution limit wrapper support

### PR DESCRIPTION
**Pull request by shazbert**
Fixes # (issue)

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and
also consider improving test coverage whilst working on a certain feature or package.

- [ ] go test ./... -race
- [ ] golangci-lint run
- [ ] Test X

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: exchanges/btcmarkets/btcmarkets_wrapper.go [hunk 1]
```diff
@@ -1091,3 +1099,33 @@ func (b *BTCMarkets) GetHistoricCandlesExtended(ctx context.Context, p currency.
 func (b *BTCMarkets) GetServerTime(ctx context.Context, _ asset.Item) (time.Time, error) {
 	return b.GetCurrentServerTime(ctx)
 }
+
+// UpdateOrderExecutionLimits sets exchange executions for a required asset type
+func (b *BTCMarkets) UpdateOrderExecutionLimits(ctx context.Context, a asset.Item) error {
+	if !a.IsValid() || a != asset.Spot {
```
Comment (gloriousCode) [inline exchanges/btcmarkets/btcmarkets_wrapper.go:1105]: If only spot is supported, is there a need for this double-check?

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Nice, tested and its all working! This should also fix updating tradable pairs 🤙 Shame that they don't provide a max-price in their API when they clearly have one:
image

File: exchanges/btcmarkets/btcmarkets_wrapper.go [hunk 2]
```diff
@@ -1091,3 +1099,33 @@ func (b *BTCMarkets) GetHistoricCandlesExtended(ctx context.Context, p currency.
...
+	}
+
+	markets, err := b.GetMarkets(ctx)
+	if err != nil {
+		return err
+	}
+
+	limits := make([]order.MinMaxLevel, len(markets))
+	for x := range markets {
+		pair, err := currency.NewPairFromStrings(markets[x].BaseAsset, markets[x].QuoteAsset)
```
Comment (gloriousCode) [inline exchanges/btcmarkets/btcmarkets_wrapper.go:1116]: nano-nit, err redeclared inside of loop

File: exchanges/btcmarkets/btcmarkets_wrapper.go (hunk 1 above)
Comment (shazbert) [reply exchanges/btcmarkets/btcmarkets_wrapper.go:1105]: nope, good pick up!

File: exchanges/btcmarkets/btcmarkets_wrapper.go (hunk 2 above)
Comment (shazbert) [reply exchanges/btcmarkets/btcmarkets_wrapper.go:1116]: Rude

**Review by gloriousCode — APPROVED** [review]
tACK! Thanks for making those changes

File: exchanges/btcmarkets/btcmarkets_types.go [hunk 3]
```diff
@@ -10,12 +10,12 @@ import (
...
-	BaseAsset      string  `json:"baseAsset"`
-	QuoteAsset     string  `json:"quoteAsset"`
+	BaseAsset      string  `json:"baseAssetName"`
+	QuoteAsset     string  `json:"quoteAssetName"`
 	MinOrderAmount float64 `json:"minOrderAmount,string"`
 	MaxOrderAmount float64 `json:"maxOrderAmount,string"`
-	AmountDecimals int64   `json:"amountDecimals,string"`
-	PriceDecimals  int64   `json:"priceDecimals,string"`
+	AmountDecimals float64 `json:"amountDecimals,string"`
+	PriceDecimals  float64 `json:"priceDecimals,string"`
```
Comment (thrasher-) [inline exchanges/btcmarkets/btcmarkets_types.go:18]: Whilst you're here, could you please add the missing field "Status"

**Review by thrasher- — CHANGES_REQUESTED** [review]
tACK, minus one extra thing I found when testing

**Review by thrasher- — APPROVED** [review]
Thanks!

**Pull request merged**